c7n/policy.py

```diff
@@ -63,71 +64,89 @@ def load(options, path, format='yaml', validate=True):
         if errors:
             raise Exception("Failed to validate on policy %s \n %s" % (errors[1], errors[0]))
 
-    return PolicyCollection(data, options)
+    collection = PolicyCollection(data, options)
+    return collection
 
 
 class PolicyCollection(object):
 
     log = logging.getLogger('c7n.policies')
 
-    def __init__(self, data, options):
+    def __init__(self, data, options, policies=None):
         self.data = data
```

Comment:
Given the new code, it is unnecessary and a little misleading to save self.data as an instance attribute.

It is only used in `__init__`.  All the calls to PolicyCollection from within the class can be changed to pass `None` as it is always ignored when passing in the optional third param.  

It is misleading because when adding together PolicyCollections we add the policies but do not add the data, so those will get out of alignment.  And given how we construct the PolicyCollection in `commands.py:_load_policies`, we will always have a blank `data` anyways.